account_invoice_ubl_peppol/models/account_move.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2021 Sunflower IT (<https://sunflowerweb.nl>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from lxml import etree
+
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def _account_invoice_ubl_use_peppol(self):
+        """ Returns True when we should use PEPPOL """
+        domain = self.company_id.get_ubl_domain_peppol()
+        domain.append(('id', '=', self.id))
+        if self.search(domain):
```

Comment:
You can use filtered_domain for this
https://github.com/odoo/odoo/blob/13.0/odoo/models.py#L5346

Reply:
Thanks! I didn't know about that.